schematic/schemas/data_model_nodes.py

```diff
@@ -0,0 +1,232 @@
+from inspect import isfunction
+import networkx as nx
+from rdflib import Namespace
+from typing import Any, Dict, Optional, Text, List, Callable
+
+from schematic.schemas.data_model_relationships import (
+    DataModelRelationships
+    )
+
+from schematic.utils.schema_utils import get_label_from_display_name, get_attribute_display_name_from_label, convert_bool_to_str, parse_validation_rules
+from schematic.utils.validate_rules_utils import validate_schema_rules
+from schematic.schemas.curie import uri2curie, curie2uri
+
+
+class DataModelNodes():
+    def __init__(self, attribute_relationships_dict):
+        self.namespaces = dict(rdf=Namespace("http://www.w3.org/1999/02/22-rdf-syntax-ns#"))
+        self.data_model_relationships = DataModelRelationships()
+        self.value_relationships = self.data_model_relationships.define_value_relationships()
+        self.edge_relationships_dictionary = self.data_model_relationships.define_edge_relationships()
+        self.properties = self.get_data_model_properties(attr_rel_dict=attribute_relationships_dict)
+
+    '''    
+    def node_present(self, G, node_name):
+        if node_name in G.nodes():
+            return True
+        else:
+            return False
+    '''
+    def gather_nodes(self, attr_info: tuple) -> list:
+        """Take in a tuple containing attriute name and relationship dictionary, and find all nodes defined in attribute information.
+        Args:
+            attr_info, tuple: (Display Name, Relationships Dictionary portion of attribute_relationships dictionary)
+        Returns:
+            nodes, list: nodes related to the given node (specified in attr_info).
+        Note:
+            Extracting nodes in this fashion ensures order is preserved.
+        """
+        # retrieve a list of relationship types that will produce nodes.
+        self.node_relationships =list(self.edge_relationships_dictionary.values())
```

Comment:
This should first be defined in the init method like `self.node_relationships = None`